netplan/cli/commands/set.py

```diff
@@ -82,7 +82,26 @@ def command_set(self):
 
             # Load YAML patch, containing our update (new or deleted settings)
             tmp.seek(0, io.SEEK_SET)
-            parser.load_yaml(tmp)
+            try:
+                parser.load_yaml(tmp)
+            except (
+                    libnetplan.LibNetplanNetDefException,
+                    libnetplan.LibNetplanParserException,
+                    libnetplan.LibNetplanPermissionException,
+                    ) as e:
+                # The libnetplan parser returns error messages similar to the one below:
+                #
+                # LibNetplanException: (null):5:7: Error in network definition: address 'a' is missing /prefixlength
+                # (null)
+                #       ^
+                #
+                # Here we extract only the part describing the problem and send it to the caller
+                exception_msg = str(e)
+                first_line = exception_msg.split('\n')[0]
+                # take the last component after the split, which should be
+                # the error message we are interested in
+                error_msg = first_line.split('Error in network definition:')[-1]
+                raise type(e)(error_msg.strip())
```

Comment:
This parsing should probably be handled in libnetplan.py transparently to the caller. The exception should provide a `.msg` field, which contains the readily parsed error message for usage by the caller.